File: app/src/main/java/com/ramez/shopp/activities/RewardsActivity.java
//package com.ramez.shopp.Activities;
//
//import android.content.Intent;
//import android.os.Bundle;
//import android.view.View;
//
//import androidx.core.content.ContextCompat;
//import androidx.fragment.app.Fragment;
//import androidx.fragment.app.FragmentManager;
//import androidx.fragment.app.FragmentStatePagerAdapter;
//
//import com.ramez.shopp.Classes.MessageEvent;
//import com.ramez.shopp.Fragments.CardFragment;
//import com.ramez.shopp.Fragments.CouponsFragment;
//import com.ramez.shopp.MainActivity;
//import com.ramez.shopp.R;
//import com.ramez.shopp.databinding.ActivityRewardsBinding;
//
//import org.greenrobot.eventbus.EventBus;
//import org.greenrobot.eventbus.Subscribe;
//import org.greenrobot.eventbus.ThreadMode;
//import org.jetbrains.annotations.NotNull;
//
//public class RewardsActivity extends ActivityBase {
//    ActivityRewardsBinding binding;
//
//    @Override
//    protected void onCreate(Bundle savedInstanceState) {
//        super.onCreate(savedInstanceState);
//        binding = ActivityRewardsBinding.inflate(getLayoutInflater());
//        View view = binding.getRoot();
//        setContentView(view);
//
//        setTitle("");
//
//        binding.viewPager.setAdapter(new MyViewPagerAdapter(getSupportFragmentManager()));
//        binding.tabs.setupWithViewPager(binding.viewPager);
//        binding.tabs.setTabTextColors(ContextCompat.getColor(getActiviy(), R.color.black),
//                ContextCompat.getColor(getActiviy(), R.color.black));
//
//
//    }
//
//    class MyViewPagerAdapter extends FragmentStatePagerAdapter {
//
//        public MyViewPagerAdapter(FragmentManager fm) {
//            super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
//        }
//
//        @NotNull
//        @Override
//        public Fragment getItem(int position) {
//
//            switch (position) {
//
//                case 0:
//                    return new CardFragment();
//                case 1:
//                    return new CouponsFragment();
//
//                default:
//                    return new CardFragment();
//            }
//        }
//
//        @Override
//        public int getCount() {
//            return 2;
//        }
//
//        @Override
//        public CharSequence getPageTitle(int position) {
//            if (position == 0) return getString(R.string.Card);
//            else if (position == 1) return getString(R.string.Coupons);
//            else return "";
//        }
//    }
//
//    @Subscribe(threadMode = ThreadMode.MAIN)
//    public void onMessageEvent(@NotNull MessageEvent event) {
//
//        if (event.type.equals(MessageEvent.TYPE_main)) {
//
//            EventBus.getDefault().post(new MessageEvent(MessageEvent.TYPE_POSITION, 0));
//            Intent intent = new Intent(getActiviy(), MainActivity.class);
//            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK);
//            startActivity(intent);
//
//        }
//
//
//    }
//
//
//}

File: app/src/main/java/com/ramez/shopp/activities/RewardsActivity.kt
package com.ramez.shopp.activities

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import com.ramez.shopp.Classes.MessageEvent
import com.ramez.shopp.fragments.CardFragment
import com.ramez.shopp.fragments.CouponsFragment
import com.ramez.shopp.MainActivity
import com.ramez.shopp.R
import com.ramez.shopp.databinding.ActivityRewardsBinding
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode


class RewardsActivity : ActivityBase() {
    var binding: ActivityRewardsBinding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityRewardsBinding.inflate(layoutInflater)
        val view: View = binding!!.root
        setContentView(view)
        title = ""
        binding!!.viewPager.adapter = MyViewPagerAdapter(supportFragmentManager)
        binding!!.tabs.setupWithViewPager(binding!!.viewPager)
        binding!!.tabs.setTabTextColors(
            ContextCompat.getColor(activiy, R.color.black),
            ContextCompat.getColor(activiy, R.color.black)
        )
    }

    internal inner class MyViewPagerAdapter(fm: FragmentManager?) :
        FragmentStatePagerAdapter(fm!!, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
        override fun getItem(position: Int): Fragment {
            return when (position) {
                0 -> CardFragment()
                1 -> CouponsFragment()
                else -> CardFragment()
            }
        }

        override fun getCount(): Int {
            return 2
        }

        override fun getPageTitle(position: Int): CharSequence? {
            return if (position == 0) getString(R.string.Card) else if (position == 1) getString(R.string.Coupons) else ""
        }
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    fun onMessageEvent(event: MessageEvent) {
        if (event.type == MessageEvent.TYPE_main) {
            EventBus.getDefault().post(MessageEvent(MessageEvent.TYPE_POSITION, 0))
            val intent = Intent(activiy, MainActivity::class.java)
            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK)
            startActivity(intent)
        }
    }
}